src/AuthenticatedAPIClient/tests/AuthenticatedAPIClient.test.jsx

```diff
@@ -165,6 +165,32 @@ describe('AuthenticatedAPIClient auth interface', () => {
     expect(result.authentication).toBeUndefined();
   });
 
+  it('has method getUserProfile', () => {
+    const userProfile = {
+      email: 'test',
+      username: 'test',
+      userProfileImageUrl: 'test',
+    };
+    client.get = jest.fn();
+
+    client.get.mockReturnValueOnce(new Promise(resolve => resolve({
+      data: userProfile,
+    })));
+    client.getUserProfile()
+      .then((result) => {
+        expect(client.get).toHaveBeenCalled();
+        expect(result).toBe(userProfile);
+      });
+
+    const message = 'Failed to get user profile.';
+    // eslint-disable-next-line prefer-promise-reject-errors
+    client.get.mockReturnValueOnce(new Promise((resolve, reject) => reject({ message })));
```

Comment:
`Promise.reject({ message })` would avoid having to new up a Promise and also let you remove the eslint disable